src/lua/datetime.lua

```diff
@@ -429,25 +427,48 @@ local function parse_tzoffset(str)
     return offset[0]
 end
 
-local function datetime_new_raw(epoch, nsec, tzoffset)
+--[[
+    Parse timezone name similar way as datetime_parse_full parse
+    full literal.
+]]
+local function parse_tzname(tzname)
+    check_str(tzname, 'parse_tzname()')
+    local ptzindex = ffi.new('int16_t[1]')
```

Comment:
I guess it's better to use ffi stash here
https://github.com/tarantool/tarantool/commit/a5549a44db835a59ad6ab82d206172f9d8047a2d

Reply:
[You need to write github bot - which would scan all pull-requests and if there is `ffi.new` discovered then would suggest to use stash. Or may be you have already done so? ;) ]

```diff
diff --git a/src/lua/datetime.lua b/src/lua/datetime.lua
index 28dc6f175..12e1ebab2 100644
--- a/src/lua/datetime.lua
+++ b/src/lua/datetime.lua
@@ -164,6 +164,10 @@ local date_dt_stash = buffer.ffi_stash_new('dt_t[1]')
 local date_dt_stash_take = date_dt_stash.take
 local date_dt_stash_put = date_dt_stash.put
 
+local date_int16_stash = buffer.ffi_stash_new('int16_t[1]')
+local date_int16_stash_take = date_int16_stash.take
+local date_int16_stash_put = date_int16_stash.put
+
 local datetime_t = ffi.typeof('struct datetime')
 local interval_t = ffi.typeof('struct interval')
 
@@ -433,13 +437,19 @@ end
 ]]
 local function parse_tzname(tzname)
     check_str(tzname, 'parse_tzname()')
-    local ptzindex = ffi.new('int16_t[1]')
-    local ptzoffset = ffi.new('int16_t[1]')
+    local ptzindex = date_int16_stash_take()
+    local ptzoffset = date_int16_stash_take()
     local len = builtin.tnt_datetime_parse_tz(tzname, #tzname, ptzoffset,
                                               ptzindex)
     if len > 0 then
-        return ptzoffset[0], ptzindex[0]
-    elseif len == -builtin.TZ_NYI then
+        local tzoffset, tzindex = ptzoffset[0], ptzindex[0]
+        date_int16_stash_put(ptzoffset)
+        date_int16_stash_put(ptzindex)
+        return tzoffset, tzindex
+    end
+    date_int16_stash_put(ptzoffset)
+    date_int16_stash_put(ptzindex)
+    if len == -builtin.TZ_NYI then
         error(("could not parse '%s' - nyi timezone"):format(tzname))
     elseif len == -builtin.TZ_AMBIGUOUS then
         error(("could not parse '%s' - ambiguous timezone"):format(tzname))
```